src/txn/txn.c

```diff
@@ -440,11 +440,37 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "read_timestamp", 0, &cval));
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
+		bool round_to_oldest;
+		int cmp;
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
+		WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
+		    cmp = __wt_timestamp_cmp(&ts,
+		    &txn_global->oldest_timestamp));
+		if (cmp < 0) {
+			/*
+			 * If given read timestamp is earlier than oldest
+			 * timestamp then round the read timestamp to
+			 * oldest timestamp based on round_to_oldest option.
+			 */
+			WT_RET(__wt_config_gets_def(session, cfg,
+			    "round_to_oldest", 0, &cval));
+			round_to_oldest = cval.val;
```

Comment:
A new variable 'round_to_oldest' is not requires in the function. In the if condition below you can directly use the 'cval.val', like this "if (cval.val) {". This way an additional variable and an assignment can be avoided. That would  also be consistent with other cval usages in the code.